### PR TITLE
Add speed end points

### PR DESCRIPTION
**Pull request by yokha**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-rest/models/
+rest/models/model_*
+rest/openapi.yaml
```
Comment (Ekberg) [inline .gitignore]: The `rest/openapi.yaml` should be committed and not ignored

File: rest/client.go [hunk 2]
```diff
@@ -27,6 +27,10 @@ type Client interface {
 	GetRelatedComponents(context.Context, models.Relation) ([]models.RelatedComponent, error)
 	CreateComponentRelation(context.Context, models.Relation, uuid.UUID) error
 	DeleteComponentRelation(context.Context, models.Relation, uuid.UUID) (err error)
+
+	GetComponentSpeed(context.Context, uuid.UUID) (models.GetComponentSpeedResponse, error)
```
Comment (Ekberg) [inline rest/client.go]: The client method should return something else than the response. We usually extract what the consumer is interested in. Perhaps this method should return a configuration and a calculated speed?

File: rest/client.go [hunk 3]
```diff
@@ -27,6 +27,10 @@ type Client interface {
...
+	SetComponentSpeed(context.Context, uuid.UUID, models.SpeedConfiguration) (models.PutComponentSpeedResponse, error)
```
Comment (Ekberg) [inline rest/client.go]: This should not return any response imo

File: rest/client.go [hunk 4]
```diff
@@ -92,11 +96,9 @@ func (c *client) DeleteComponent(ctx context.Context, id uuid.UUID) error {
 		Assign("component", id).
 		SetHeader("Accept", "application/json")
 
-	if _, err := c.Do(ctx, request); err != nil {
-		return err
-	}
+	_, err := c.Do(ctx, request)
```
Comment (karlssonerik) [inline rest/client.go]: Isn't it cleaner to keep the return err on a separate line from return nil? If only that stack traces are easier to debug?

**Review by Ekberg — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (yokha) [reply .gitignore]: what's is the reason behind committing it if we get it it from asset-components service?

File: rest/client.go (hunk 2 above)
Comment (yokha) [reply rest/client.go]: I can do that . just extract both to 2 output.

File: rest/client.go (hunk 4 above)
Comment (yokha) [reply rest/client.go]: it's more code preference, I can revert it

File: rest/client.go (hunk 2 above)
Comment (yokha) [reply rest/client.go]: fixed

File: rest/client.go (hunk 3 above)
Comment (yokha) [reply rest/client.go]: fixed

File: rest/client.go (hunk 4 above)
Comment (yokha) [reply rest/client.go]: fixed

File: .gitignore (hunk 1 above)
Comment (Ekberg) [reply .gitignore]: Same reason why we keep a go.lock file. We should have a reproducable branch where we get the same results if we rebuild today or in two weeks.

Comment (yokha) [reply .gitignore]: recovered

File: Makefile [hunk 5]
```diff
@@ -21,8 +21,8 @@ rest/models/: rest/openapi.yaml
 			--global-property models,modelDocs=false \
 			--generator-name go \
 			--additional-properties packageName=models \
-			--output $@
+			--output $(shell dirname $@)
```
Comment (Ekberg) [inline Makefile]:
```suggestion
			--output $(@D)
```

**Review by Ekberg — DISMISSED** [review]

File: Makefile (hunk 5 above)
Comment (Ekberg) [reply Makefile]: Minor, resolve if you don't want to do it.

**Review by Ekberg — APPROVED** [review]

**Pull request merged**